packages/ingest/test/test-HttpsProviderClient.js

```diff
@@ -212,3 +236,33 @@ test('HttpsProviderClient supports basic auth with redirects for sync', async (t
     await recursivelyDeleteS3Bucket(destinationBucket);
   }
 });
+
+test('HttpsProviderClient supports basic auth for sync with redirect to different host/port', async (t) => {
```

Comment:
So this unit test covers the gap which our logic wasn't handling where during a redirect to a site with a different host/port, basic auth credentials were not forwarded, so the request would fail with a 401 error. This behavior comes from `got` internally:

https://github.com/sindresorhus/got/blob/main/source/core/index.ts#L744

This scenario is exactly what happens when trying to download a file from a DAAC HTTP provider: you request a file from their host (e.g. lpdaac.com/file.hdf) and you get redirected to Earthdata login (e.g. urs.earthdata.nasa.gov) to provide credentials before you are redirected back to download the file. For example you will see the redirects in action if you visit this URL in an incognito window:

https://gpm1.gesdisc.eosdis.nasa.gov/data/GPM_L3/GPM_3IMERGHH.06/2020/001/3B-HHR.MS.MRG.3IMERG.20200101-S223000-E225959.1350.V06B.HDF5

Given the above, I also wonder if we need an integration test. Because the `got` behavior to disable credentials on redirects to different hosts is a recent change that probably broke our sync with basic auth functionality and we didn't catch it because our unit tests didn't test that scenario. With this new unit test, in theory we're covered, but I'm still wondering. However, it would be pretty tricky to set up an integration test that enforces a basic auth redirect to a different host and it's yet another test to maintain.

Reply:
Yeah, this is tricky. My opinion is that we should _not_ have an integration test for this. If a third-party library introduces a breaking change, ideally we would have tests to catch it but I'm not sure we want to write tests that explicitly test the behavior of those libraries. That sounds like what an integration test would be:

`Redirects to protected files work if auth is provided` (or something)

Before your change above, if we had that test it would have felt to me like testing the library (e.g. redirects include auth headers) pretty directly. Now that we have a custom handler it's a little different but ...is it? That handler is tested here at the unit level. Actually handling the redirect still seems like what an integration test would test and that doesn't seem like something we would want to include.